### PR TITLE
NPF: add Direct Debit popup

### PR DESCRIPTION
**Pull request by joelochlann**
Direct Debit popup will now appear when submitting the payment form with Direct Debit selected.

Popup itself (closing, entering details, progressing through the form, etc) works, but we don't yet submit the actual payment details at the end. That'll be in the next PR.

### CODE REVIEW

File: assets/components/directDebit/directDebitPopUpForm/directDebitPopUpForm.scss [hunk 1]
```diff
@@ -74,9 +74,10 @@
   height: 40px;
   width: 40px;
   float: right;
+  display: flex;
```
Comment (joelochlann) [inline assets/components/directDebit/directDebitPopUpForm/directDebitPopUpForm.scss]: The fact that the cross is correctly centered on the existing pages is just a random consequence of the font-size of the container, and it wasn't centered on the new page. So I've used `flex` to center the button, everywhere

File: app/services/HttpIdentityService.scala [hunk 2]
```diff
@@ -204,7 +204,7 @@ class HttpIdentityService(apiUrl: String, apiClientToken: String)(implicit wsCli
     ) { resp =>
         resp.json.validate[GuestRegistrationResponse]
           .asEither
-          .bimap( _.mkString(","), response => UserIdWithGuestAccountToken.fromGuestRegistrationResponse(response))
+          .bimap(_.mkString(","), response => UserIdWithGuestAccountToken.fromGuestRegistrationResponse(response))
```
Comment (joelochlann) [inline app/services/HttpIdentityService.scala]: IntelliJ did this against my will

File: assets/pages/new-contributions-landing/components/ContributionForm.jsx [hunk 3]
```diff
@@ -238,12 +244,18 @@ function ContributionForm(props: PropTypes) {
...
+          callback={() => Promise.resolve()}
+          isPopUpOpen={props.isDirectDebitPopUpOpen}
+        />
       </div>
     );
 }
 
 ContributionForm.defaultProps = {
   error: null,
+  isDirectDebitPopUpOpen: false,
```
Comment (joelochlann) [inline assets/pages/new-contributions-landing/components/ContributionForm.jsx]: not sure if I need this when the prop is linked to Redux state?

File: assets/pages/new-contributions-landing/contributionsLanding.scss [hunk 4]
```diff
@@ -163,21 +166,21 @@ body {
   font: normal 16px/1.5 $gu-text-egyptian-web;
 }
 
-h1 {
+.header {
```
Comment (joelochlann) [inline assets/pages/new-contributions-landing/contributionsLanding.scss]: `h1` is too general!

File: assets/pages/new-contributions-landing/contributionsLanding.scss [hunk 5]
```diff
@@ -163,21 +166,21 @@ body {
...
- 
+
 input,
 select,
 textarea,
 button {
   font: inherit;
 }
-  
+
```
Comment (joelochlann) [inline assets/pages/new-contributions-landing/contributionsLanding.scss]: IntelliJ being dogmatic, here and elsewhere

File: assets/components/directDebit/directDebitPopUpForm/directDebitPopUpForm.scss (hunk 1 above)
Comment (regiskuckaertz) [reply assets/components/directDebit/directDebitPopUpForm/directDebitPopUpForm.scss]: I use flex _all the time™_

File: assets/pages/new-contributions-landing/components/ContributionForm.jsx (hunk 3 above)
Comment (regiskuckaertz) [reply assets/pages/new-contributions-landing/components/ContributionForm.jsx]: only when it is optional, i.e. `?boolean` or `boolean | void`.

File: assets/components/directDebit/directDebitPopUpForm/directDebitPopUpForm.scss (hunk 1 above)
Comment (joelochlann) [reply assets/components/directDebit/directDebitPopUpForm/directDebitPopUpForm.scss]: yep... although I just discovered some caveats which affect browsers we support. and this specific case, where I'm applying `display: flex` to a `<button></button>`. https://github.com/philipwalton/flexbugs#9-some-html-elements-cant-be-flex-containers

Comment (joelochlann) [reply assets/components/directDebit/directDebitPopUpForm/directDebitPopUpForm.scss]: Fix on the way....

File: app/services/HttpIdentityService.scala (hunk 2 above)
Comment (jranks123) [reply app/services/HttpIdentityService.scala]: Haha this kept happening to me too!

File: assets/pages/new-contributions-landing/components/ContributionForm.jsx [hunk 6]
```diff
@@ -68,6 +70,8 @@ type PropTypes = {|
   onThirdPartyPaymentDone: Token => void,
   checkoutFormHasBeenSubmitted: boolean,
   setCheckoutFormHasBeenSubmitted: () => void,
+  openDirectDebitPopUp: void => void,
```
Comment (jranks123) [inline assets/pages/new-contributions-landing/components/ContributionForm.jsx]: The usual pattern we use is `() => void`. Is there a reason for using `void => void`?

Comment (joelochlann) [reply assets/pages/new-contributions-landing/components/ContributionForm.jsx]: no! thanks for pointing out, it should be `() => void`

File: app/services/HttpIdentityService.scala (hunk 2 above)
Comment (joelochlann) [reply app/services/HttpIdentityService.scala]: yep, it's enraging... hopefully this will put the issue to bed

**Pull request merged**